src/main/java/frc/robot/commands/CrocodileDefaultCommand.java

```diff
@@ -3,33 +3,49 @@
 import java.util.function.BooleanSupplier;
 import java.util.function.DoubleSupplier;
 
+import edu.wpi.first.math.filter.Debouncer;
+import edu.wpi.first.wpilibj.XboxController;
 import edu.wpi.first.wpilibj2.command.CommandBase;
 import frc.robot.subsystems.CrocodileSubsystem;
 
 public class CrocodileDefaultCommand extends CommandBase {
     private final CrocodileSubsystem subsystem;
-    private final DoubleSupplier motorSpeed;
-    private final BooleanSupplier slowMovement;
+    private final DoubleSupplier trigger;
+    private final XboxController driverController;
+    private boolean beamBreakTracker = false;
+    private final Debouncer debouncer;
 
-    public CrocodileDefaultCommand(CrocodileSubsystem subsystem, DoubleSupplier motorSpeed,
-            BooleanSupplier slowMovement) {
+    public CrocodileDefaultCommand(CrocodileSubsystem subsystem, DoubleSupplier trigger) {
         this.subsystem = subsystem;
-        this.motorSpeed = motorSpeed;
-        this.slowMovement = slowMovement;
+        this.trigger = trigger;
+        //TODO: get driver controller port
+        //controller to use for rumble
+        driverController = new XboxController(0);
```

Comment:
We don't typically give commands full, unfettered access to the game controllers. We usually pass lambdas to provide access to the pieces of the game controller we need. I recommend doing that here. It would keep you consistent with all the other places we do this, and ensure we always know in one place what we're doing with our controllers.

I think what you're looking for here is `DoubleConsumer`.

So, consider take a parameter something like `DoubleConsumer giveHapticFeedback` and when you want to give 'em a shake, do `giveHapticFeedback.accept(0.5);`. When you want to stop, do `giveHapticFeedback.accept(0);`.

Then, when you create the default command, just like when you pass in other lambdas to give access to sticks and such, pass this:

```
            (t) -> driverController.getHID().setRumble(RumbleType.kBothRumble, t),
```

Reply:
Thanks, I believe I have a solution here!